apps/src/templates/SwipePrompt.jsx

```diff
@@ -0,0 +1,138 @@
+/* global navigator */
+import React from 'react';
+import PropTypes from 'prop-types';
+import {connect} from 'react-redux';
+import cookies from 'js-cookie';
+import {dismissSwipeOverlay} from '@cdo/apps/templates/arrowDisplayRedux';
+import trackEvent from '@cdo/apps/util/trackEvent';
+
+const styles = {
+  overlay: {
+    position: 'absolute',
+    zIndex: 1,
+    opacity: '90%'
+  },
+  minecraft: {
+    top: '62px',
+    left: '17px'
+  }
+};
+
+const HideSwipeOverlayCookieName = 'hide_swipe_overlay';
+
+export class SwipePrompt extends React.Component {
+  static propTypes = {
+    useMinecraftStyling: PropTypes.bool,
+
+    // from redux
+    buttonsAreVisible: PropTypes.bool.isRequired,
+    buttonsAreDisabled: PropTypes.bool.isRequired,
+    hasBeenDismissed: PropTypes.bool.isRequired,
+    onDismiss: PropTypes.func.isRequired,
+    dismissAction: PropTypes.string.isRequired
+  };
+
+  constructor(props) {
+    super(props);
+    if (this.swipeOverlayOverrideSet()) {
+      cookies.remove(HideSwipeOverlayCookieName, {path: '/'});
+    }
+  }
+
+  componentDidUpdate(prevProps) {
+    const {hasBeenDismissed, dismissAction} = this.props;
+    if (
+      hasBeenDismissed &&
+      !prevProps.hasBeenDismissed &&
+      !this.hideOverlayCookieSet()
+    ) {
+      // The overlay was just dismissed. Don't show it again for a while.
+      cookies.set(HideSwipeOverlayCookieName, 'true', {expires: 30, path: '/'});
+      trackEvent('Research', 'HideSwipeOverlay', 'hide-' + dismissAction);
+    }
+  }
+
+  swipeOverlayOverrideSet() {
+    window.location.search.indexOf('force_show_swipe_overlay') !== -1;
+  }
+
+  touchSupported() {
+    // Note: if you are planning to use this logic in other places, you should
+    // know that this is not foolproof for detecting touch support. It is
+    // impossible to detect touch support with 100% certainty.
+    // http://www.stucox.com/blog/you-cant-detect-a-touchscreen/
+    'ontouchstart' in window || navigator.maxTouchPoints;
+  }
+
+  hideOverlayCookieSet() {
+    return (
+      cookies.get(HideSwipeOverlayCookieName) && !this.swipeOverlayOverrideSet
+    );
+  }
+
+  render() {
+    const {
+      onDismiss,
+      buttonsAreVisible,
+      buttonsAreDisabled,
+      hasBeenDismissed,
+      useMinecraftStyling
+    } = this.props;
+
+    if (
+      this.hideOverlayCookieSet() ||
+      hasBeenDismissed ||
```

Comment:
is `hasBeenDismissed` redundant with `this.hideOverlayCookieSet()`?

Reply:
Thanks for catching that! It is redundant here.

Reply:
Ok. I'm realizing why I left `hasBeenDismissed` here. Turns out there are two reasons:
1. `this.hideOverlayCookieSet()` is important on page load, but since it doesn't change the state/props, this component won't re-render if the the cookie changes. 
2. We override the cookie if the user has `force_show_swipe_overlay` in the browser. `hasBeenDismissed` allows a dev to feel like they have the same functionality as the cookie without listening for the cookie